internal/exec/engine.go

```diff
@@ -272,7 +273,20 @@ func (e *Engine) fetchReferencedConfig(cfgRef types.ConfigReference) (types.Conf
 	if err != nil {
 		return types.Config{}, err
 	}
-	rawCfg, err := e.Fetcher.FetchToBuffer(*u, resource.FetchOptions{})
+	var headers http.Header
+	if cfgRef.HTTPHeaders != nil && len(cfgRef.HTTPHeaders) > 0 {
+		if u.Scheme != "http" && u.Scheme != "https" {
+			e.Logger.Crit("Cannot use HTTP headers with %v source scheme", u.Scheme)
+			return types.Config{}, fmt.Errorf("cannot use HTTP headers with %v source scheme", u.Scheme)
```

Comment:
I think these checks should live in the `Validate()` functions of their associated type (i.e. the `Validate()` for `ConfigReference`, the one for `FileContents`, and the one for `CaReference`). That way, we do this type of validation up front. We can also dedupe the error string by putting it in `shared/errors/errors.go`.

Reply:
Done